src/hotspot/share/gc/shared/workgroup.cpp

```diff
@@ -391,18 +384,11 @@ void SubTasksDone::all_tasks_completed_impl(uint n_threads,
     assert(task_index < _n_tasks, "Array in range.");
     assert(!_tasks[task_index], "%d is both claimed and skipped.", task_index);
   }
+}
 #endif
-  uint observed = _threads_completed;
-  uint old;
-  do {
-    old = observed;
-    observed = Atomic::cmpxchg(&_threads_completed, old, old+1);
-  } while (observed != old);
-  // If this was the last thread checking in, clear the tasks.
-  uint adjusted_thread_count = (n_threads == 0 ? 1 : n_threads);
-  if (observed + 1 == adjusted_thread_count) {
-    clear();
-  }
+
+bool SubTasksDone::valid() {
+  return _tasks != NULL;
```

Comment:
This function should can never return false, since _tasks is initialized in the constructor and the value deleted in the destructor.  It should be removed and any callers fixed.

Reply:
Removed.